### PR TITLE
Self hosted agents

### PR DESCRIPTION
**Pull request by lucaspin**

### CODE REVIEW

**Comment by shiroyasha** [comment]
Looks good so far.

One important part that is not yet covered is how to use self-hosted agents in your pipelines.

File: docs/ci-cd-environment/configuring-a-self-hosted-agent.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+### `disconnect-after-job`
+
+By default, the agent does not disconnect from Semaphore and shuts down after completing a job. If you want to do that instead, set `disconnect-after-job` to true.
+
+### `shutdown-hook-path`
+
+By default, nothing fancy is executed when the agent shuts down. This parameter accepts a path to a bash script in the host to be executed once that happens. It can be useful to perform cleaning up operations (pushing the agent logs to some external storage, shutting down the machine).
+
+It can also be useful when used in conjunction with `disconnect-after-job` in order to rotate the agents and make sure you get a clean one for every job you run.
```
Comment (shiroyasha) [inline docs/ci-cd-environment/configuring-a-self-hosted-agent.md:95]: Give an example hook here. For example, a file that contains `sudo poweroff -f`.

Comment (shiroyasha) [reply docs/ci-cd-environment/configuring-a-self-hosted-agent.md:95]: Document also what happens if the file does not exist.

Comment (lucaspin) [reply docs/ci-cd-environment/configuring-a-self-hosted-agent.md:95]: Done.

**Comment by lucaspin** [comment]
> One important part that is not yet covered is how to use self-hosted agents in your pipelines.

oh, right, forgot about that 😅. Just added it 🙇‍♂️

**Comment by shiroyasha** [comment]
From my side, it looks 👌 

I would like to hear your thoughts as well @AleksandarCole and @csidyel.

File: docs/ci-cd-environment/using-self-hosted-agent-types.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+## Using a self hosted agent type in your pipelines
+
+In order to use a self hosted agent type in your pipelines, the machine type on the agent definition of your pipeline needs to be updated:
+
+```diff
+agent:
+  machine:
+-   os_image: ubuntu1804
+-   type: e1-standard-2
++   type: s1-<your-self-hosted-agent-type-name>
```
Comment (shiroyasha) [inline docs/ci-cd-environment/using-self-hosted-agent-types.md:25]: It is better to use a concrete example here, with a sentence like:

> To use a self-hosted agent type in your pipelines, the machine type on the agent definition of your pipeline needs to be updated. For example, if your self-hosted machine type is `s1-linux-small`, use this configuration:

```
agent:
  machine:
-   os_image: ubuntu1804
-   type: e1-standard-2
+   type: s1-linux-small
```

**Review by csidyel — APPROVED** [review]
LGTM

**Comment by markoa** [comment]
- The slug of the file translates into URL so please change them to include less characters, at least with a `a`, eg. `configure/install-self-hosted-agents.md`
- I notice multiple mentions of "Semaphore 2.0", the standard way to refer the product is "Semaphore".
- "self hosted agent" is not grammatically correct, it should be "self-hosted agent" (many instances)

**Pull request merged**